src/exchange-connector/exchangeDirectoryParser.ts

```diff
@@ -52,9 +53,7 @@ export function extractFiles(
           );
         })
       );
-    }
-  });
+    });
 
-  // The then here collapses the return from an array of void to a single void
   return Promise.all(promises).then();
```

Comment:
`.then()` here is unnecessary - it just clones the promise, which we don't need to do.